core/src/main/scala/org/apache/spark/rpc/RpcEnv.scala

```diff
@@ -264,9 +264,15 @@ object RpcTimeout {
     require(timeoutPropList.nonEmpty)
 
     // Find the first set property or use the default value with the first property
-    val foundProp = timeoutPropList.view.map(x => (x, conf.getOption(x))).filter(_._2.isDefined).
-      map(y => (y._1, y._2.get)).headOption.getOrElse(timeoutPropList.head, defaultValue)
-
+    val itr = timeoutPropList.iterator
+    var foundProp =  (timeoutPropList.head,defaultValue)
+    while (itr.hasNext){
+      val propKey = itr.next()
+      conf.getOption(propKey) match {
+        case Some(prop) => foundProp = (propKey,prop)
```

Comment:
this logic is a little different than the original -- it'll take the *last* set property, rather than the first.